base_layer/wallet/src/wallet.rs

```diff
@@ -166,9 +166,10 @@ where
         key_manager_backend: TKeyManagerBackend,
         shutdown_signal: ShutdownSignal,
         master_seed: CipherSeed,
-        wallet_type: WalletType,
+        wallet_type: Option<WalletType>,
         user_agent: String,
     ) -> Result<Self, WalletError> {
+        let wallet_type = read_or_create_wallet_type(wallet_type, &wallet_database)?;
```

Comment:
What's the benefit doing it on start instead of initialization? 

FFI wasn't calling the same initialization function?

Reply:
the init was in console wallet, not being called by the wallet_lib and wallet_ffi